### DISCUSSION TITLE
Ruff doesnt pick up TODO:

### DISCUSSION THREAD

**Discussion by MH-ABE** [astral-sh/ruff#10975] [Q&A] [answered]
Hi, I'm probably just doing something stupid, but in VScode Ruff (or pylance for that matter) doesn't pick up TODO: comments, which they used to for me some time ago. It looks like this:

![image](https://github.com/astral-sh/ruff/assets/126556032/018da1c3-fc34-403a-abe9-44355b183635)

ruff check --show-settings, dint show anything that relates to TODO from what I could see

Vs code 1.87.2
Ruff 0.3.1, v2024.16.0 using the bundled version

Init lines from the output:
```
2024-04-16 14:51:16.292 [info] Global settings: {
    "cwd": "C:\\Users\\1820134\\AppData\\Local\\Programs\\Microsoft VS Code",
    "workspace": "C:\\Users\\1820134\\AppData\\Local\\Programs\\Microsoft VS Code",
    "path": [],
    "ignoreStandardLibrary": true,
    "interpreter": [],
    "importStrategy": "fromEnvironment",
    "codeAction": {
        "fixViolation": {
            "enable": true
        },
        "disableRuleComment": {
            "enable": true
        }
    },
    "lint": {
        "enable": true,
        "run": "onType",
        "args": [
            "--line-length=120",
            "--target-version=py312"
        ]
    },
    "format": {
        "args": []
    },
    "enable": true,
    "organizeImports": true,
    "fixAll": true,
    "showNotifications": "off"
}
```

I hope someone can help me diagnose this.
Thanks in advance!

**Comment by MichaReiser** [comment]
Hi. 

I'm not sure I understand what the expected output is. Do you expect lint errors for todo comments? Can you share a specific example (or say which line in your screenshot should be flagged)? What's your Ruff configuration?

**Comment by MH-ABE** [comment]
Hi MichaResier, thanks.
Yeah, so under the heading https://docs.astral.sh/ruff/faq/#which-tools-does-ruff-replace this one is listed: https://pypi.org/project/flake8-todos/ 
So for example I'm expecting "T005: missed text in TODO." to be caught on line 1.
But really I'd like the existence of a task comment to be listed as a warning too, but maybe that's not standard?

Perhaps I have other problems too, as I can't get line-length E501 to trigger either. Still other warnings and errors work fine.

Only explicit configs that I can find that I have put in is:
```
"ruff.lint.args": [
    "--line-length=120",
    "--target-version=py312",
],
```
I installed ruff into my env so now I have 0.3.7:
```
2024-04-17 09:08:29.657 [info] Found ruff 0.3.7 at c:\myenvpath\Scripts\ruff.exe
2024-04-17 09:08:29.658 [info] Running Ruff with: c:\...\Scripts\ruff.exe ['check', '--force-exclude', '--no-cache', '--no-fix', '--quiet', '--output-format', 'json', '-', '--line-length=120', '--target-version=py312', '--stdin-filename', 'c:\\myfilenamepath\\Arbetskortprint.py']
```

ruff --show-settings : [output](https://github.com/astral-sh/ruff/files/15007434/ruff.--show-settings.txt)

**Reply by MichaReiser** [reply, marked as answer]
You have to enable said rules. They aren't part of the default rule set. See [https://docs.astral.sh/ruff/linter/#rule-selection]. 

In your case, you want `--extend-select=E501 --extend-select=TD` (I think)

**Reply by MH-ABE** [reply]
Ow wow, that was easy, thanks for the quick help! Was indeed I who did something stupid...

Perhaps to help somebody else out there, in vscodes settings.json you need to config it like below, the .toml way as per the docs naturally doesn't work in json.
```
"ruff.lint.args": [
    "--select=E,F,UP,B,SIM,I,TD,FIX",  // this is how you get multiple codes. TD and FIX for TODOS. FIX002 is the "just fix it" warning.
],
```
